samples/dfu/nrf9160/src/main.c

```diff
@@ -0,0 +1,159 @@
+/*
+ * Copyright (c) 2022 INTELLINIUM <giuliano.franchetto@intellinium.com>
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#include <zephyr/kernel.h>
+#include <zephyr/dfu/mcuboot.h>
+#include <modem/nrf_modem_lib.h>
+#include <net/fota_download.h>
+#include <modem/lte_lc.h>
+#include <dfu/dfu_target_uart.h>
+#include <zephyr/device.h>
+
+/**
+ * @typedef update_start_cb
+ *
+ * @brief Signature for callback invoked when button is pressed.
+ */
+typedef void (*update_start_cb)(void);
+
+/**
+ * @brief Initialization struct.
+ */
+
+static struct k_work fota_work;
+static update_start_cb update_start;
+static char filename[128] = {0};
+
+struct update_sample_init_params {
+	/* Callback invoked when the button is pressed */
+	update_start_cb update_start;
+
+	/* Update file to download */
+	const char *filename;
+};
+
+static void fota_work_cb(struct k_work *work)
+{
+	ARG_UNUSED(work);
+
+	if (update_start != NULL) {
+		update_start();
+	}
+}
+
+static void modem_configure(void)
+{
+	BUILD_ASSERT(!IS_ENABLED(CONFIG_LTE_AUTO_INIT_AND_CONNECT),
+		     "This sample does not support auto init and connect");
+	int err;
+
+	printk("LTE Link Connecting ...\n");
```

Comment:
nrf5340 is using LOG_x macros and this is using printk, can we standardise on using LOG_x?